vscode-extension/src/util.ts

```diff
@@ -301,3 +301,22 @@ export async function getPythonPath(): Promise<string> {
     pythonExtension.exports.settings.getExecutionDetails().execCommand[0];
   return pythonPath;
 }
+
+/**
+ * Checks if the specified Python interpreter is version 3.10 or higher.
+ * @param pythonPath The path to the Python interpreter.
+ * @returns A promise that resolves to a boolean indicating if the version is >= 3.10.
+ */
+
+export function isPythonVersionAtLeast310(pythonPath: string): boolean {
+  try {
+    // Use Python to check compatible version
+    const command = `${pythonPath} -c "import sys; print(sys.version_info >= (3, 10))"`;
```

Comment:
Why this command vs just doing `--version` check? IMO that would be cleaner so that we can do the comparison in code ourselves

Reply:
Actually, we already should have that version from stdout above, right? I don't think we need to run any python here at all

Reply:
executing `python --version` returns a literal like `Python 3.11.7`. 

Felt like parsing this into a version number and then comparing was trickier than using python itself to validate version. 

I can update if that would be easier to follow

Reply:
Ah, ok. I guess there are some nuances with it as well (e.g. comparator of version numbers). I'm fine with keeping as-is using python to check